pkg/ibmcloudprovider/volume_provider.go

```diff
@@ -164,3 +193,67 @@ func (icp *IBMCloudStorageProvider) GetConfig() *config.Config {
 func (icp *IBMCloudStorageProvider) GetClusterInfo() *utils.ClusterInfo {
 	return icp.ClusterInfo
 }
+
+// UpdateAPIKey ...
+func (icp *IBMCloudStorageProvider) UpdateAPIKey(logger *zap.Logger) error {
+	logger.Info("Updating API key in cloud storage provider")
+	vpcBlockConfig := &vpcconfig.VPCBlockConfig{
+		VPCConfig:    icp.ProviderConfig.VPC,
+		IKSConfig:    icp.ProviderConfig.IKS,
+		APIConfig:    icp.ProviderConfig.API,
+		ServerConfig: icp.ProviderConfig.Server,
+	}
+	vpcAPIKey := vpcBlockConfig.VPCConfig.G2APIKey
+
+	if icp.ProviderConfig.IKS != nil && (icp.ProviderConfig.IKS.Enabled || os.Getenv("IKS_ENABLED") == "True") && icp.ProviderConfig.VPC.Encryption {
+		apiKeyImp, err := utils.NewAPIKeyImpl(logger)
+		if err != nil {
+			logger.Error("Unable to create API key getter", zap.Reflect("Error", err))
+			return err
+		}
+		logger.Info("Created NewAPIKeyImpl...")
+		err = apiKeyImp.UpdateIAMKeys(icp.ProviderConfig)
+		if err != nil {
+			logger.Error("Unable to get API key", local.ZapError(err))
+			return err
+		}
+		// If the retrieved API key is the same as previous one, returning error
+		if vpcAPIKey == icp.ProviderConfig.VPC.G2APIKey {
+			logger.Error("API key is not reset")
+			return errors.New("API key is not found, reset the api key. If reset, retry after few minutes")
```

Comment:
keep this error msg https://github.com/IBM/ibm-csi-common/blob/master/pkg/messages/messages_en.go and use it